predicators/envs/kitchen.py

```diff
@@ -0,0 +1,275 @@
+"""A Kitchen environment wrapping kitchen from https://github.com/google-
+research/relay-policy-learning."""
+import copy
+from typing import Any, Dict, List, Optional, Sequence, Set
+
+import matplotlib
+import numpy as np
+from gym.spaces import Box
+
+try:
+    from mujoco_kitchen.kitchen_envs import OBS_ELEMENT_INDICES
+    from mujoco_kitchen.utils import make_env
+except ImportError:
+    pass
+from predicators import utils
+from predicators.envs import BaseEnv
+from predicators.settings import CFG
+from predicators.structs import Action, EnvironmentTask, Image, Object, \
+    Observation, Predicate, State, Type, Video
+
+
+class KitchenEnv(BaseEnv):
+    """Kitchen environment wrapping dm_control Kitchen."""
+
+    gripper_type = Type("gripper", ["x", "y", "z"])
+    object_type = Type("obj", ["x", "y", "z", "angle"])
+
+    def __init__(self, use_gui: bool = True) -> None:
+        super().__init__(use_gui)
+
+        # Predicates
+        self._At, self._OnTop, self._On = self.get_goal_at_predicates()
```

Comment:
since we usually think of "On" as on top, can we rename it to something like "DialTurned"?